source/agora/node/admin/AdminInterface.d

```diff
@@ -183,10 +189,9 @@ public class AdminInterface : NodeControlAPI
     public override string encryptionKeyQR (
         string app, ulong height, out string contentType, out string vary)
     {
-        // TODO: PreImage to be released at that height
-        // Or it can be pre-released, so it can be `height + PreimageRevealPeriod`
-        PreImageInfo pre_image;
-
+        // Gets the pre-image of the requested height.
+        // It can be pre-released, so it add to the `Reveal period + 1`
+        Hash pre_image = this.enroll_man.cycle[Height(height + this.enroll_man.PreimageRevealPeriod + 1)];
```

Comment:
```suggestion
        PreImageInfo preimage;
        this.enroll_man.getNextPreimage(preimage, [Height(height - 1)]);
```
and how about using `preimage.hash` afterward.
If it does not suit for your purpose, I would recommend a new function in `EM`

Reply:
The getNextPreimage function can only import pre-images from an enrolled enrollment cycle range.
So I created a function to get a pre-image of the block height.

Reply:
@Geod24 ping